dashboard/app/controllers/levels_controller.rb

```diff
@@ -23,6 +23,10 @@ def index
   # GET /levels/1
   # GET /levels/1.json
   def show
+    if @level.properties["pages"]
```

Comment:
small refinement, if you change this (and the corresponding in ScriptLevelsController) to @level.try(:pages) you can hide from the controller that the implementation of pages is in properties.